### PR TITLE
lib: bin: lwm2m_carrier: Add oper_id check to lwm2m_os_sec_psk_delete()

### PR DESCRIPTION
**Pull request by stig-bjorlykke**
This delete from lwm2m_carrier library is only intended for
specific operators. Add a check for this.

Signed-off-by: Stig Bjørlykke <stig.bjorlykke@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM but @rlubos and @lemrey  should approve before we merge

**Review by rlubos — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Pull request merged**
